chanfitness/chaneventstore.go

```diff
@@ -448,14 +459,20 @@ func (c *ChannelEventStore) getChanInfo(req channelInfoRequest) (*ChannelInfo,
 func (c *ChannelEventStore) recordFlapCount() error {
 	flapCounts := make(map[route.Vertex]*channeldb.FlapCount)
 
+	// Get our current time so that we can decide whether to cooldown our
+	// flap count for each peer with the same value.
+	now := c.cfg.Clock.Now()
+
 	for peer, monitor := range c.peers {
 		flapCount, lastFlap := monitor.getFlapCount()
 		if lastFlap == nil {
 			continue
 		}
 
 		flapCounts[peer] = &channeldb.FlapCount{
-			Count:    uint32(flapCount),
+			Count: uint32(
+				cooldownFlapCount(now, flapCount, *lastFlap),
```

Comment:
iiuc this won't really cooldown the flapcount as we always get it from the peer monitor where the flapcount always increases so the max we could get is 0.9 * flapcount

Reply:
Moved this up and tested with some regtest nodes. Since we only need the cooldown for rate limiting, decided to keep it as we need it as you suggest. 

It is the case that we may cooldown then increment, but that makes sense to me in the following case:
- peer with 100 flap count connects, time since flap > cooldown period
- cooldown to 95
- update to 96, since they have just flapped, also update their last flapped